scipy/spatial/transform/rotation.pyx

```diff
@@ -216,9 +216,9 @@ cdef double[:, :] _compute_euler_from_matrix(
                           "all angles.")
 
     # Reverse role of extrinsic and intrinsic rotations, but let third angle be
-    # zero for gimbal locked cases
+    # zero for gimbal locked cases. Take a copy, to enforce contiguous memory layout.
     if extrinsic:
-        angles = angles[:, ::-1]
+        angles = angles[:, ::-1].copy()
```

Comment:
Instead of flipping the entire array, just swap `_angels[0]` with `_angles[2]` at the end of the loop above.

Reply:
Like 
```
angles[:, 0], angles[:, 2]  = angles[:, 2], angles[:, 0]
```
It should work too. 

I have no strong preference between the two variants.

Reply:
The comment there says, that the third element is set to zero "for gimbal locked cases."
So I think we cannot do it that way. The tests also fail, if I do so. Let's stick to copying it -- or do you have concerns about performance? I will add the regression test right away

Reply:
I did some investigation:

1. The comment is obsolete. I'm not sure now whether it was obsolete from the beginning (i.e. it was supposed to be implemented as in the comment, but never actually was) or a change was introduced at some point making the code and the comment contradict each other. My guess it was like this from the very beginning. I will check it.
2. The suggested implementation `angles[:, 0], angles[:, 2] = angles[:, 2], angles[:, 0]` doesn't work because `angles[:, 0]` are overwritten by `angles[:, 2]` first and then are used to to overwrite `angles[:, 2]`, i. e. `angles[:, 2]` doesn't change. I guess it was somewhat expected.

Overall - `angles[:, ::-1].copy()` is perfect.

Reply:
No, I'm suggesting putting _inside_ the loop:
```python
if extrinsic:
    _angles[0], _angles[2] = _angles[2], _angles[0]
```

Reply:
Makes sense.

Do you think it is necessary though? I think that converting to Cython has already hurted readability quite a bit, I would suggest sticking to vectorized operations where not performance critical.

But generally don't have a strong preference. So @peterbell10 should we do it as you suggest?

Reply:
Is that really harder to read? I would think most people find vectorized operations more abstract.

Reply:
I think yes, in a sense that these are small details spread over code in a loop. 

Vectorized operations are higher level and can be understood and thought as a single operation. In this function we first inverse `seq` array and in the end inverse `angles` (if `intrinsic`). For me personally it easier to follow the logic of a function this way. 

I suggest that @marscher decide on this point, if you don't mind.

Reply:
I've chosen to to invert inside of the loop, since all other operations are also done there and have to be understood as well when digging through the algorithm.
Thank you all for your feedback!